src/ert/scheduler/scheduler.py

```diff
@@ -57,38 +63,32 @@ def __init__(
         self.driver = driver
         self._tasks: MutableMapping[int, asyncio.Task[None]] = {}
 
-        self._jobs: MutableMapping[int, Job] = {
-            real.iens: Job(self, real) for real in (realizations or [])
+        self._jobs: Mapping[int, Job] = {
+            real.iens: Job(real, on_complete=self._on_job_complete)
```

Comment:
Looks good! But (just our of curiosity mostly), why not to to use:
`self._tasks[iens].add_done_callback(self._on_job_complete)`?

Reply:
Because it's not when the task is done running, it's when the job is `COMPLETED` successfully.

Reply:
Yes, but we can check in `_on_job_complete` that `self.state == COMPLETED`?